lib/galaxy/datatypes/binary.py

```diff
@@ -1224,6 +1224,56 @@ def display_peek(self, dataset):
             return f"Binary Anndata file ({nice_size(dataset.get_size())})"
 
 
+class Grib(Binary):
+    """
+    Class describing an GRIB file
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('test.grib')
+    >>> Grib().sniff(fname)
+    True
+    >>> fname = get_test_fname('interval.interval')
+    >>> Grib().sniff(fname)
+    False
+    """
+    file_ext = "grib"
+    # GRIB not yet in EDAM (work in progress). For now, so set to binary
+    edam_format = "format_2333"
+
+    def __init__(self, **kwd):
+        super().__init__(**kwd)
+        self._magic = b'GRIB'
+
+    def sniff(self, filename):
+        # The first 4 bytes of any GRIB file are GRIB
+        try:
+            _uint8struct = struct.Struct(b">B")
+            f = open(filename, 'rb')
+            header = f.read(4)
+            if header == self._magic:
+                tmp = f.read(4)
+                edition = _uint8struct.unpack_from(tmp, 3)[0]
```

Comment:
```suggestion
    def sniff_prefix(self, sniff_prefix):
        # The first 4 bytes of any GRIB file are GRIB
        try:
            if sniff_prefix.startswith_bytes(self._magic):
                tmp = contents_header_bytes[:4]
                _uint8struct = struct.Struct(b">B")
                edition = _uint8struct.unpack_from(tmp, 3)[0]
```
sniff_prefix is a small optimization that reads in the first portion of a file once, and then provides this to all sniff_prefix methods. This way we don't have to open the files to be sniffed N times for N datatypes that define a sniffer function.